Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Backend/WebRequests/RavelWebRequest.cs

```diff
@@ -300,11 +312,22 @@ public class TokenWebRequest : RavelWebRequest {
         /// <param name="api">Api addition part, added between base url and version of the whole url.</param>
         /// <param name="version">Version addition, added after api part of the url.</param>
         public TokenWebRequest(Method method, string api, string version) : base(method, api, version) {
-            AddHeader("Authorization", "Bearer " + GetToken());
+            AddToken();
         }
 
         public TokenWebRequest(string api, string version, WWWForm form) : base(api, version, form) {
-            AddHeader("Authorization", "Bearer " + GetToken());
+            AddToken();
+        }
+
+        private void AddToken() {
+            string token = GetToken();
+            
+            if (string.IsNullOrEmpty(token)) {
+                _error = InternalError.MissingToken;
+                return;
+            }
+            
+            AddHeader("Authorization", "Bearer " + token);
         }
```

Comment:
Delete unused duplicate  public RavelWebRequest(string url, WWWForm form, Dictionary<string, string> parameters = null, Dictionary<string, string> header = null)

Reply:
This webrequest is used in Ravel (non-creator). I've kept the webrequest files the same, so we can keep them in sync.